cmapi/cmapi_server/helpers.py

```diff
@@ -343,67 +343,75 @@ def broadcast_new_config(
-                r.raise_for_status()
-            except requests.Timeout as e:
-                logging.warning(
-                    f'Timeout while pushing new config to "{node}"'
+                async with session.put(
+                    url, headers=headers, json=body, ssl=False, timeout=120
+                ) as response:
+                    response.raise_for_status()
+                logging.info(f'Node "{node}" config put successfull.')
+            except aiohttp.ClientResponseError as err:
+                message = (
+                    f'Node "{node}" config put failed with status: '
+                    f'"{err.status}" and message: {err.message}'
                 )
-            except Exception as e:
-                logging.warning(
-                    f'Got an unexpected error pushing new config to "{node}"',
-                    exc_info=True
+                logging.error(message)
+                raise CMAPIBasicError(message)
+            except aiohttp.ClientError as err:
+                message = (
+                        f'Node "{node}" config put failed with ClientError: '
+                        f'{str(err)}'
+                    )
+                logging.error(message)
+                raise CMAPIBasicError(message)
+            except asyncio.TimeoutError:
+                message = f'Node "{node}" config put failed by Timeout: '
+                logging.error(message)
+                raise CMAPIBasicError(message)
+            except Exception as err:
+                message = (
+                    f'Node "{node}" config put failed by undefined exception: '
+                    f'{str(err)}'
                 )
-            else:
-                success_nodes.append(node)
-                success = True
-                break
-        if not success:
-            failed_nodes.append(node)
+                logging.error(message)
+                raise CMAPIBasicError(message)
 
     loop = asyncio.new_event_loop()
     asyncio.set_event_loop(loop)
     tasks = [
-        update_config(node, success_nodes, failed_nodes, headers, body)
+        update_config(node, headers, body)
         for node in nodes
     ]
-    loop.run_until_complete(asyncio.wait(tasks))
+    error_messages = []
+    results = loop.run_until_complete(
+        asyncio.gather(*tasks, return_exceptions=True)
+    )
+    for result in results:
+        if isinstance(result, Exception):
+            error_messages.append(result.message)
     loop.close()
-
-    if len(success_nodes) > 0:
-        logging.info(
-            f'Successfully pushed new config file to {success_nodes}'
-        )
-    if len(failed_nodes) > 0:
-        logging.error(
-            f'Failed to push the new config to {failed_nodes}'
+    if error_messages:
+        errors_str = ', '.join(error_messages)
+        final_message = (
+            f'Broadcasting config to nodes failed with errors: {errors_str}'
```

Comment:
Nice!

Reply:
That would be helpful to have nodes ids here together with the corresponding errors.

Reply:
For sure. It's already done at lines [:371](https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3400/files#diff-f1f673533210d56c8ffc9ede947d9211040737f97dd02dec79c1dfe513d60aa3R371) [:378](https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3400/files#diff-f1f673533210d56c8ffc9ede947d9211040737f97dd02dec79c1dfe513d60aa3R378)    [:384](https://github.com/mariadb-corporation/mariadb-columnstore-engine/pull/3400/files#diff-f1f673533210d56c8ffc9ede947d9211040737f97dd02dec79c1dfe513d60aa3R384)